tests/unit/Operation/AllCollectionTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+use loophp\collection\Collection;
+use tests\loophp\collection\CollectionTestCase;
+
+/**
+ * @internal
+ * @covers \loophp\collection\Operation\All
+ */
+final class AllCollectionTest extends CollectionTestCase
+{
+    public function dataProvider()
+    {
+        yield [
+            [1, 2, 3],
+            [1, 2, 3],
+            false,
+        ];
```

Comment:
```suggestion
    /**
    * @return Generator<array{0: array<mixed>, 1: array<mixed>, 2: bool}>
    */
    public function dataProvider(): Generator
    {
        yield [
            [1, 2, 3],
            [1, 2, 3],
            false,
        ];
```

the types are missing here but I think it's because the test paths are ignored in static analysis. I would say since we're doing this from scratch again it would be helpful to analyse the test namespace as well. Tests effectively show how the code is being used by users, so static analysis will help spot any issues from that point of view as well

Reply:
Oh! should we test the types in PHPUnit ?

Reply:
That's the thing to do if your life feel too easy :sweat_smile: 
Seriously, these things hold no value.

Reply:
Hmm I think the value comes from getting the types right on the source code itself 😄. If you look at the closed PRs you will see what I mean :).

Anyway we can add that in a separate PR, so don't worry worry about it for now @drupol